FFTGeneratorConfigs.scala

```diff
@@ -0,0 +1,23 @@
+package fftgenerator
+
+import freechips.rocketchip.config.{Field, Parameters, Config}
+
+//cparameter to enable FFT
+case object FFTEnableKey extends Field[Option[FixedTailParams]](None)
+
+class WithFFTGenerator (enable: Boolean = true) extends Config((site, here, up) => {
+  case FFTEnableKey => Some(FixedTailParams())
+})
+
+// parameter to set FFT mmio registers base address
+case object FFTBaseAddrKey extends Field[Int](0x2000)
+
+class WithFFTBaseAddr (baseAddress: Int) extends Config((site, here, up) => {
+  case FFTBaseAddrKey => baseAddress
+})
+
+case object FFTNumPoints extends Field[Int](2)
+
+class WithFFTNumPoints (numPoints: Int) extends Config((site, here, up) => {
+  case FFTNumPoints => numPoints
```

Comment:
You should do this:

`val config = p(FFTEnableKey).get.copy(n = p(FFTNumPoints), lanes = p(FFTNumPoints))`

here instead of in the other location. For example, look at this code to set the BootROM: https://github.com/ucb-bar/chipyard/blob/02adf86b8261ca4b5baf35b3257e15cdbabd7f87/generators/chipyard/src/main/scala/ConfigFragments.scala#L39-L41. It copies the previous set parameters and modifies a small portion of it (the file name). You can do something similar with `case NameOfKey => up(NameOfKey, site).copy(mychange=hi)` which will give you access to the parameters to modify.